docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc

```diff
@@ -0,0 +1,69 @@
+= Importing code to App Studio
+
+[role="_abstract"]
+After you click **Create an application**, the **Grab some code** page opens. From here, choose one of the following source code options to begin building your application:
+
+* Import your own code.
+* Start with an {ProductName} code sample.
+
+== Importing code from your GitHub repository
+
+To use your own code to build your application, complete the following steps:
+
+. Enter the link to your repository in the **Git repository URL** field. {ProductName} verifies your URL right away. You don’t even have to press **Enter**.
+. Optional: Add a **Git reference** to point to code in a specific branch, tag, or commit that you want to use to build your application.
+. Optional: Indicate a **Context directory** to specify the subdirectory for the source code you want to use. 
+
+[NOTE]
+====
+Make sure this path is correct. You might get an error if your build context folder is your root directory but your Dockerfile is in a subdirectory of that folder.
+====
+[start=4]
+. Click **Import code**. 
+
+[TIP]
+====
+If {ProductName} can’t access your Git repository, see <<Granting App Studio access to your Git repository>> for help.
+====
+
+{ProductName} clones, scans, and analyzes your Git repository to determine how many components to add to your application and which runtime to use. The types of files in your repository, and also your Git directory file hierarchy, determine the number and type of components to add. For example, {ProductName} looks for devfiles and Dockerfiles, and also language-specific build configuration files like pom.xml for Java or package.json for Node.js.
+
+[NOTE]
+====    
+{ProductName} can build applications that are written in the following language stacks and frameworks:
+
+    * Python
+    * Java
+    * Go
+    * .NET
+    * Node.js
+    * Quarkus
+    * Spring Boot
+====
+
+== Using a code sample
+// ADD LINK TO CHLOE'S CONTENT!
+
+. Optional: In the code sample **Search** field, begin typing the name of the code you want to work with. If {ProductName} has code samples in that language stack, the code sample results filter accordingly.
+. Optional: If you see a language stack or framework that you want to work with, click either its **Open Git repository** or **Import sample** link. Follow the resulting instructions to explore or test whatever language and framework you choose.   
+* If you click **Open Git repository**, a new tab opens to the GitHub repository for the code framework you chose.
+* If you click **Import sample**, your application  **Overview** tab opens. From here, you can monitor your continuous integration/continuous deployment (CI/CD) activity.
```

Comment:
Import sample is the main path here. They can open the sample git repo, but there aren't any "resulting instructions" to see, they can just decide if that is something they want to start from, or they could take the opportunity to fork that sample to their GitHub account, and then use the new URL to bring that code.

Reply:
my "directive" is to focus on the "import code" procedures. would you argue that a) seeing the route once deployed, b) what to do if no route displays ...  should be doc'd for Summit? @mattreid bc if so, i can add!

can't users monitor build progress from the Overview tab (per last bullet)?

Reply:
Yes, if they look at the visualization on the Overview tab, they should see some basic information and status and hopefully can use that to find what they want next. I'm wondering if we want to suggest more specific next actions that connect back to why they're importing in the first place and thinking about next questions. I imported, how do I see what I built? When and where does that get deployed? When does my code turn into an app that I can do something with? What should I do if it doesn't import successfully? I'm not sure how that fits or doesn't fit into what should be covered here in this module, but that's where my head goes. Their end goal isn't just to import and watch it spin, that's the precursor to more interesting stuff.

Reply:
@mattreid you're entirely right, ofc! let me finish this "importing code" stuff, get it merged and pubbed, and come right back to this. i'll create Jiras for this content, too, and add you as a contributor. is this OK?

Reply:
@mattreid I added content about resolving failed import issues. the rest we'll add after summit, i'll create jiras for them, etc.